ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java

```diff
@@ -63,9 +73,15 @@ private static void validateMLInput(Input input) {
             throw new IllegalArgumentException("Input should be MLInput");
         }
         MLInput mlInput = (MLInput) input;
-        DataFrame dataFrame = mlInput.getDataFrame();
-        if (dataFrame == null || dataFrame.size() == 0) {
-            throw new IllegalArgumentException("Input data frame should not be null or empty");
+        MLInputDataset inputDataset = mlInput.getInputDataset();
+        if (inputDataset == null) {
+            throw new IllegalArgumentException("Input data set should not be null");
+        }
+        if (inputDataset instanceof DataFrameInputDataset) {
```

Comment:
Good catch, I think I missed throw exception part for this case. Thanks, will fix

Reply:
Sorry I didn't look the code when reply your comment. Actually this method is to validate ML input. The input data set could be not `DataFrameInputDataset`, for example it could be `SearchQueryInputDataset` which will query data from index. We already have validation logic in `SearchQueryInputDataset` , so no need to recheck here. So this part should be ok.